desc/vmec_utils.py

```diff
@@ -468,3 +482,451 @@ def vmec_boundary_subspace(eq, RBC=None, ZBS=None, RBS=None, ZBC=None):  # noqa:
+    )
+    # add the (0,0) mode to modes_sin if eq.sym to abide by booz xform convention
+    # even though the mode is trivially zero for sin(mt-nz)
+    if eq.sym:
+        modes_sin = np.insert(modes_sin, 0, [-1, 0, 0], axis=0)
+
+    ## 1D Arrays
+    keys_1d = [
+        "I",
+        "G",
+    ]
+    # this should be compressed then to just the radial profiles
+    grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, rho=r_half, NFP=NFP, sym=eq.sym)
+    data_1d = eq.compute(keys_1d, grid=grid)
+    jlist = file.createVariable("jlist", np.int32, ("comput_surfs",))
+    jlist.long_name = (
+        "1-based radial indices of the surfaces for which the "
+        "transformation to Boozer coordinates was computed. 2 corresponds to the "
+        "first half-grid point."
+    )
+    jlist.units = "None"
+    jlist[:] = np.arange(1, s_half.size + 1) + 1
+
+    ixm_b = file.createVariable("ixm_b", np.int32, ("mn_modes",))
+    ixm_b.long_name = (
+        "Poloidal mode numbers m for which the Fourier amplitudes rmnc,"
+        "bmnc etc are stored"
+    )
+    ixm_b.units = "None"
+    ixm_b[:] = modes[:, 1] if eq.sym else modes[inds_cos, 1]
+
+    ixn_b = file.createVariable("ixn_b", np.int32, ("mn_modes",))
+    ixn_b.long_name = (
+        "Toroidal mode numbers n for which the Fourier amplitudes"
+        "rmnc, bmnc etc are stored"
+    )
+    ixn_b.units = "None"
+    ixn_b[:] = modes_sin[:, 2] * eq.NFP if eq.sym else modes[inds_cos, 2] * eq.NFP
+
+    iotas = file.createVariable("iota_b", np.float64, ("radius",))
+    iotas.long_name = (
+        "Rotational transform. The radial grid corresponds to the"
+        "requested surfaces, and a 0 is prepended"
+    )
+    iotas.units = "None"
+
+    iotas[0:] = np.insert(
+        -grid.compress(eq.compute("iota", grid=grid, data=data_1d)["iota"]), 0, 0
+    )
+
+    buco_b = file.createVariable("buco_b", np.float64, ("radius",))
+    buco_b.long_name = (
+        "Coefficient multiplying grad theta_Boozer in the covariant"
+        "representation of the magnetic field vector, often denoted I(psi)."
+    )
+    buco_b.units = "None"
+    buco_b[0] = 0
+
+    buco_b[1:] = -grid.compress(data_1d["I"])
+
+    bvco_b = file.createVariable("bvco_b", np.float64, ("radius",))
+    bvco_b.long_name = (
+        "Coefficient multiplying grad zeta_Boozer in the covariant"
+        "representation of the magnetic field vector, often denoted G(psi)."
+    )
+    bvco_b.units = "None"
+    bvco_b[0] = 0
+
+    bvco_b[1:] = grid.compress(data_1d["G"])
+
+    # TODO: assuming this is on full mesh
+    presf = file.createVariable("pres_b", np.float64, ("radius",))
+    presf.long_name = "pressure on full mesh"
+    presf.units = "Pa"
+    presf[:] = eq.compute("p", grid=LinearGrid(rho=r_full, theta=0, zeta=0))["p"]
+
+    beta = file.createVariable("beta_b", np.float64, ("radius",))
+    beta.long_name = "Blank, only included for compatibility reasons"
```

Comment:
is this our choice or is it also like this in the simsopt and fortran versions?

Reply:
simsopt has it as zeros, fortran doesnot. I will change to compute this correctly